lib/chef/provider/machine_image.rb

```diff
@@ -20,49 +19,73 @@ def new_driver
     @new_driver ||= run_context.chef_provisioning.driver_for(new_resource.driver)
   end
 
+  def chef_managed_entry_store
+    @chef_managed_entry_store ||= Provisioning.chef_managed_entry_store(new_resource.chef_server)
+  end
```

Comment:
Could there be a common superclass or module mixin hiding with the `#chef_managed_entry_store` methods in the providers? I'm not sure either way, but at least the impls are the same--which is relieving